flink-runtime/src/test/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceUtilsTest.java

```diff
@@ -668,7 +668,7 @@ private void validateFail(final Configuration config) {
 		try {
 			TaskExecutorResourceUtils.resourceSpecFromConfig(config);
 			fail("Configuration did not fail as expected.");
-		} catch (Throwable t) {
+		} catch (IllegalConfigurationException | IllegalArgumentException t) {
```

Comment:
I think ideally we want all the exceptions we throw from the resource configuring calculations are `IllegalConfigurationException` with meaningful messages.

I can see that without catching `IllegalArgumentException` here, we will have test cases failure, which indicates that when the improper configurations do happen users will not get meaningful messages.

I've caught and converted `IllegalArgumentException` into meaningful `IllegalConfigurationException` for constructing `MemorySize`. Maybe we can do the same for`RangeFraction` that causes the test failures.